onnxscript/function_libs/torch_lib/graph_building.py

```diff
@@ -402,6 +402,13 @@ def torch_graph(self):
     def initializers(self) -> Mapping[str, torch.Tensor]:
         return self._initializers
 
+    # NOTE: This setter is used in torch converter when we activate fake mode,
+    #       we need to filter out the initializers that has fake tensor. This
+    #       is because we don't want to introduce fake tensor in onnxscript.
+    @initializers.setter
```

Comment:
Is add_initializer not enough?

Reply:
We could filter fake tensor INSIDE `add_initializer`, but do you think we should bring fake tensor into it?

Reply:
I suggests filtering outside of the call if possible. The exporter should be responsible for deciding which initializer to add

Reply:
We want it to be initializers, but we don't want it to have default values.

Reply:
So a tensor proto w/o data?

Reply:
Constant tensor is not an input and we want to keep its value. The reason that it's not going through `_wrap_fx_args_as_onnxscript_args` is because ORT prefers constant tensors to be initializers. As for others, buffers and weights under fake mode, those values would become fake tensor which further crashes during `to_model_proto` if we don't detach the fake tensor from value.

Reply:
I see. I am actually a little confused on how initializers are handled. Happy to chat more tomorrow.

Reply:
Specifically 

        # Mapping from intializer name to data(torch.Tensor).
        self._initializers: Dict[str, torch.Tensor] = {}
        # Mapping from intializer name to input(TorchScriptTensor).
        self._initializers_inputs: Dict[str, TorchScriptTensor] = {}

@bowenbao

Reply:
Should we have this by 8/25 to catch 2.1?

Reply:
Changes in pytorch need to catch the 8/25 cut.

Reply:
@justinchuby  here is an example of what these three initializer name mapped values are and used.


        initializer_data: value in self._initializers
        initializer_input: value in self._initializers_inputs
        initializer_input_from_parent: value in self._initializers_inputs_from_parent

```
MainGraph(...)
  %initializer_input_from_parent = onnx::Constant(value_t=initializer_data)
  %sub_graph_output = onnx::SubGraph(%initializer_input_from_parent)

SubGraph(%initializer_input)
  %output = onnx::Op(%initializer_input, ...)
```

Reply:
All the complexity is for building the symbolic connection of subgraph input and maingraph initializer. We can probably refactor this away and move connection logic to exporter, such that the logic for dismissing fake tensor initializer data look more natural, but the deadline is too tight.